rosco-manifests/src/main/java/com/netflix/spinnaker/rosco/manifests/helmfile/HelmfileBakeManifestService.java

```diff
@@ -0,0 +1,49 @@
+package com.netflix.spinnaker.rosco.manifests.helmfile;
+
+import static com.netflix.spinnaker.rosco.manifests.BakeManifestRequest.TemplateRenderer;
+
+import com.google.common.collect.ImmutableSet;
+import com.netflix.spinnaker.kork.artifacts.model.Artifact;
+import com.netflix.spinnaker.rosco.jobs.BakeRecipe;
+import com.netflix.spinnaker.rosco.jobs.JobExecutor;
+import com.netflix.spinnaker.rosco.manifests.BakeManifestEnvironment;
+import com.netflix.spinnaker.rosco.manifests.BakeManifestService;
+import java.io.IOException;
+import java.util.Base64;
+import org.springframework.stereotype.Component;
+
+@Component
+public class HelmfileBakeManifestService extends BakeManifestService<HelmfileBakeManifestRequest> {
+  private final HelmfileTemplateUtils helmfileTemplateUtils;
+  private static final ImmutableSet<String> supportedTemplates =
+      ImmutableSet.of(TemplateRenderer.HELMFILE.toString());
+
+  public HelmfileBakeManifestService(
+      HelmfileTemplateUtils helmTemplateUtils, JobExecutor jobExecutor) {
+    super(jobExecutor);
+    this.helmfileTemplateUtils = helmTemplateUtils;
+  }
+
+  @Override
+  public Class<HelmfileBakeManifestRequest> requestType() {
+    return HelmfileBakeManifestRequest.class;
+  }
+
+  @Override
+  public boolean handles(String type) {
+    return supportedTemplates.contains(type);
+  }
+
+  public Artifact bake(HelmfileBakeManifestRequest helmfileBakeManifestRequest) throws IOException {
```

Comment:
This is so close to [HelmBakeManifestService](https://github.com/spinnaker/rosco/blob/d49a43507428ebd67bd0942f1050c755e45ab6bc/rosco-manifests/src/main/java/com/netflix/spinnaker/rosco/manifests/helm/HelmBakeManifestService.java).  How can we share the code?

Reply:
I would put this code within the new package suggested here `com.netflix.spinnaker.rosco.manifests.utils.helm`. what do you think?

Reply:
Looking at the code this is a function a bit hard to reduce in an OOP. I would need to create a separate class (abstract) to share only thing function. happy to hear here any proposals.